go/ql/test/experimental/CWE-522-DecompressionBombs/test.go

```diff
@@ -0,0 +1,301 @@
+	http.Handle("/Decompress", DecompressHandler)
+	err := http.ListenAndServe(":8080", nil)
+	if err != nil {
+		return
+	}
+
+}
+func DecompressHandler(w http.ResponseWriter, request *http.Request) {
+	TarDecompressor(request.Body, "gz")
+	ZipNewReader(request.Body)
+	ZipNewReader2(request.Body)
+	ZipOpenReader(request.FormValue("filepathba"))
+	ZipOpenReaderSafe(request.PostFormValue("test"))
+	GZipOpenReaderSafe(request.PostFormValue("test"))
+	GZipReader(request.Body, "dest")
+}
+
+func GZipOpenReaderSafe(filename string) {
+	var src io.Reader
+	src, _ = os.Open(filename)
+	gzipR, _ := gzip.NewReader(src)
+	dstF, _ := os.OpenFile("./test", os.O_RDWR|os.O_CREATE|os.O_TRUNC, 0755)
+	defer dstF.Close()
+	var newSrc io.Reader
+	newSrc = io.LimitReader(gzipR, 1024*1024*1024*5)
+	_, _ = io.Copy(dstF, newSrc)
+}
+
+func ZipOpenReaderSafe(filename string) {
+	r, _ := zip.OpenReader(filename)
+	var totalBytes int64 = 0
+	for _, f := range r.File {
+		rc, _ := f.Open()
+		for {
+			result, _ := io.CopyN(os.Stdout, rc, 68)
+			if result == 0 {
+				_ = rc.Close()
+				break
+			}
+			totalBytes = totalBytes + result
+			if totalBytes > 1024*1024 {
+				fmt.Print(totalBytes)
+				break
+			}
+		}
+	}
+}
+
+func GZipReader(src io.Reader, dst string) {
+	gzipR, _ := gzip.NewReader(src)
+	dstF, _ := os.OpenFile(dst, os.O_RDWR|os.O_CREATE|os.O_TRUNC, 0755)
+	defer dstF.Close()
+	var newSrc io.Reader
+	newSrc = io.Reader(gzipR)
+	_, _ = io.Copy(dstF, newSrc)
+}
+
+func ZipOpenReader(filename string) {
+	// Open the zip file
+	r, _ := zip.OpenReader(filename)
+	for _, f := range r.File {
+		rc, _ := f.Open()
+		for {
+			result, _ := io.CopyN(os.Stdout, rc, 68)
+			if result == 0 {
+				_ = rc.Close()
+				break
+			}
+			fmt.Print(result)
+			_ = rc.Close()
+		}
+	}
+	rKlauspost, _ := zipKlauspost.OpenReader(filename)
+	for _, f := range rKlauspost.File {
+		rc, _ := f.Open()
+		for {
+			result, _ := io.CopyN(os.Stdout, rc, 68)
+			if result == 0 {
+				_ = rc.Close()
+				break
+			}
+			fmt.Print(result)
+			_ = rc.Close()
+		}
+	}
+}
+
+func ZipNewReader(file io.Reader) {
+	file1, _ := io.ReadAll(file)
+	zipReader, _ := zip.NewReader(bytes.NewReader(file1), int64(32<<20))
+	for _, file := range zipReader.File {
+		fileWriter := bytes.NewBuffer([]byte{})
+		fileReaderCloser, _ := file.Open()
+		result, _ := io.Copy(fileWriter, fileReaderCloser)
+		fmt.Print(result)
+	}
+}
+
+func ZipNewReader2(file io.Reader) {
+	file2, _ := io.ReadAll(file)
+	zipReaderKlauspost, _ := zipKlauspost.NewReader(bytes.NewReader(file2), int64(32<<20))
+	for _, file := range zipReaderKlauspost.File {
+		fileWriter := bytes.NewBuffer([]byte{})
+		// file.OpenRaw()
+		fileReaderCloser, _ := file.Open()
+		result, _ := io.Copy(fileWriter, fileReaderCloser)
+		fmt.Print(result)
+	}
+}
+func serve(w http.ResponseWriter, r *http.Request) {
+	if r.Body != nil {
+		if data, err := ioutil.ReadAll(r.Body); err == nil {
+			fmt.Println(data)
+		}
+	}
+}
```

Comment:
There are no results in this function, and it isn't called from anywhere. Can it be deleted?